layers/gpuav/spirv/inject_conditional_function_pass.cpp

```diff
@@ -123,6 +123,9 @@ bool InjectConditionalFunctionPass::Instrument() {
                 continue;  // Currently can't properly handle injecting CFG logic into a loop header block
             }
             auto& block_instructions = (*block_it)->instructions_;
+
+            block_instrumented_table_.clear();
+
             for (auto inst_it = block_instructions.begin(); inst_it != block_instructions.end(); ++inst_it) {
                 // Every instruction is analyzed by the specific pass and lets us know if we need to inject a function or not
                 if (!RequiresInstrumentation(*function, *(inst_it->get()))) {
```

Comment:
if not `RequiresInstrumentation` then instrument?
```suggestion
                if (!RequiresInstrumentation(*function, *(inst_it->get()))) {
```

Reply:
what?

Reply:
```cpp
if (!RequiresInstrumentation(*function, *(inst_it->get()))) {
                    if ((*inst_it)->Opcode() == spv::OpSampledImage) {
                        const uint32_t result_id = (*inst_it)->ResultId();
                        const uint32_t type_id = (*inst_it)->TypeId();
                        const uint32_t copy_id = module_.TakeNextId();
                        function->ReplaceAllUsesWith(result_id, copy_id);
                        inst_it++;
                        (*block_it)->CreateInstruction(spv::OpCopyObject, {type_id, copy_id, result_id}, &inst_it);
                        inst_it--;
                    }
                    continue;
}
```

So you have `CreateInstruction` (instrumentation) protected by a `if not RequiresInstrumentation`, seems weird